diff --git a/Makefile b/Makefile
index da6e081..5055883 100644
--- a/Makefile
+++ b/Makefile
@@ -503,11 +503,11 @@ endif
 	$(Q)ln -sf xxhsum $(DESTDIR)$(BINDIR)/xxh32sum
 	$(Q)ln -sf xxhsum $(DESTDIR)$(BINDIR)/xxh64sum
 	$(Q)ln -sf xxhsum $(DESTDIR)$(BINDIR)/xxh128sum
-	@echo Installing man pages
-	$(Q)$(INSTALL_DATA) $(MAN) $(DESTDIR)$(MANDIR)/xxhsum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh32sum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh64sum.1
-	$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh128sum.1
+	#@echo Installing man pages
+	#$(Q)$(INSTALL_DATA) $(MAN) $(DESTDIR)$(MANDIR)/xxhsum.1
+	#$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh32sum.1
+	#$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh64sum.1
+	#$(Q)ln -sf $(MAN) $(DESTDIR)$(MANDIR)/xxh128sum.1
 	@echo xxhash installation completed
 
 .PHONY: uninstall
